src/destructors.md

```diff
@@ -1,97 +1,396 @@
 # Destructors
 
-When an [initialized]&#32;[variable] in Rust goes out of scope or a [temporary]
-is no longer needed its _destructor_ is run. [Assignment] also runs the
-destructor of its left-hand operand, unless it's an uninitialized variable. If a
-[struct] variable has been partially initialized, only its initialized fields
-are dropped.
+When an [initialized]&#32;[variable] or [temporary] goes out of
+[scope](#drop-scopes) its *destructor* is run, or it is *dropped*. [Assignment]
+also runs the destructor of its left-hand operand, if it's initialized. If a
+variable has been partially initialized, only its initialized fields are
+dropped.
 
-The destructor of a type consists of
+The destructor of a type `T` consists of:
 
-1. Calling its [`std::ops::Drop::drop`] method, if it has one.
+1. If `T: Drop`, calling [`<T as std::ops::Drop>::drop`]
 2. Recursively running the destructor of all of its fields.
-    * The fields of a [struct], [tuple] or [enum variant] are dropped in
-      declaration order. \*
-    * The elements of an [array] or owned [slice][array] are dropped from the
-      first element to the last. \*
-    * The captured values of a [closure] are dropped in an unspecified order.
+    * The fields of a [struct] or [tuple] are dropped in declaration order.
```

Comment:
Nit: tuples are not declared, right? I think I would say "The fields of a tuple are dropped in order." or something